docs/tutorials/10_torch_qgan.ipynb

```diff
@@ -0,0 +1,999 @@
HA7Nmzhdd5gsowDHQ6+Y0hesr9/vHPodUCw7DwekNISdHu3sNxHJqamtDf34+Kioq4c37I5E/5hiUmqf19PePWdUqTpKpYUCVKUv/3fwvQ3R3Es8924K23fGhtZbBp09l46qmLAEQqlfNPvhU4+VZwHIdi2ffXOvQxOamivzkVkgpEW2AuWrQIra1Sr135aJw8NUFChkeupI4JVHbHMAyOHj2qeiOPbxdSUmExWwEMRwMS7DhF04TzFJV+Yyx6/H4/3G436urqhDzjQCBdcR88CbaZgc8bAYNerKQqkVSt6n61nFTt9UJLqbUYVXJSVQgfWUmVFU4Z4wv3y+ElCu7RseWtS4EomeSgU13fyL6xPEklbBF9JYqi4iKokx3Hjh3DVVddhV//+tdYuXLlRE9nymOapCYJ401Su7q6UFdXh7lz56Krq0t4XdxBiqZpQqFJ9JQHApEFlvF1o+b1K+ClC1Fy1v8gs3BiQhe9vQxWrWpEW1sf8vO9uPjiVNxzzzwUFsY6k7Isi4MHD0Kv16OsrCwh0qhGUkkimXjx7e7qxM6dO+Ooph09yKJwdDIheXW/JCc18d9iZqYRjzxShEceUX8f6ViTc1KVC6ckBE2l41RPTw/q6+uFQkCdjsQaRLmvKkqqOBY7uoeMsSuc4kdmWRZlZWUJ/Lai38dmCsM7fJjkSmo4GCGodocDpSsuSmBGCjmj/N4Jx9NsNmPGjBmYMWOG0LknUgwjz5ePnn+GBT47BLR2AycViZVUMjQtqEhtS+M0rge0w/0x1exiNwIVIhkMqqudAGBWI6kqxaZaSqqcpIrHDoZZ7Ny5EykpKUJnJfHDP9k3NjK2jkRShbGnBkHt6OjAFVdcgZ///Oc47bTTJno6JwSmJEmd6uH+lpYWdHR0YPny5QiFQnC5IhX8pBansQpXdHHw+8PwuyrR+Mq5CPqHYADQ/JetqA7bQOecjlNveBX6cVpMamqCOPPMBng8AQB6tLWl4g9/AP7whwO45JIOvPbaWuG9pAr+REDuphifkpqZkYZVq1YJHa8OHToEq9WK7OxsZGVlJXXx1fJJDTKycP8oLKhGC+2cVLWGAGTSx4e7xR2GNH1SVRTbMBNVvRJdIiQ5tIl9VHlMlW0GozEOgkqeiTcQCa2nW/2wmwYBCvAG9bANk570jEzc9cu9sDoyY0dUC/cTlNRodb+6KltVNQSvl4XJZIU4P1K8D4Oew7G+CEEFItcdG2XBxHE1w/306Kr7ySRYXISk0nFKISd1aGhoeJ1TJuuAVEmVP8cZTMpd79RC8gBgU2kx6szJw6pVqzAwMAC3243m5mYYDAZkZWXB6XSqKqkkksrfEimKOu4Jqtvtxvr16/HTn/4UZ5111kRP54TBlCSpE4HxIKksy6K2thbhcBgrVqwATdNCWD/SLYYFRVGSwqmIwiVeWKKnPOgfQsNLZyAUllqI6Fgv/u9fH2Lp5d1ITR+fwqfVq13weOQ3ogCA/6C2lgMQIalqFfzxQivczzBSqyNx4VQ46AdFUUI1Lcdx8Hq9cLlc2Lt3r5CHl52dHVd+rBrIZFqkpKpW949v6gnZgkqZpGqF+1tbW+FyuWLSOBL1SRXvyNXZIfixsizJ/SNO5joOFlSx5uzq+7SaBuAPDoHlrAA4zM/zYMOpL8E1AFCUDkXf6kNP/Q6Yqrfh3rt/DqNZ2YtWiuh+SW1RBZJKklmH8dFHXlxxRROCQQYU5UGE1GQAyEPkuhvuWkZzklbJOlrcLIA8vpaSqp6TGo+SSnpVPL70t6ulpHq9XlRVVUGnI+VbynNSo+uyvF2zUYWkeuTZMLKx1ZRUo8kKiqKQmpqK1NRUlJSUwOfzoaurC7W1tTh6NBWA/NqJzFNPuDZDooeMsrKy45agdnd3Y/369Xj88cdx/vnnT/R0TihMWZJKUVQSCxy0MdYkNRQKobKyEhkZGZg9e7awaNM0jXA4LBBU+WIe60QlIqmBAHJTw/CHAE/Yjt5+D7wB4P9qAX8I8A/2jhtJ9XjMAMwAWBiNrQgG3QA+BzCIUCiSx9bX14eamhqUlpaOymLMYlFXKIeCejgs0RuEtHBKeteiKAp2ux12ux1z5swR8vD4NpNZWVnIzs6GzWZLWOG3WEjqlFhJVSapibQaTQa0C6dUGgKIrlO+U9zQ0BDKy8tjLI20mgaoWVDl5ORg6dKlcLvd8Ho9AJRv9OMBYptRhSWLvJZFXztjSQ1+c80ibP38fBw4mo3HLn8JL30i7AgURSFz/sXIHLbxUYKakqpT6TilZMj61luD2LixUXgQ4Tg+dScE4AMAlwj7oClOsn8dJc7QII+vWd1vUM4ZjYekaimp8ktampMqJakejwfV1dVYsmQJWDYIuXIaS1KjZFKupKZnKncx0uqSZZe1LhXftozG2GvCYrEIaRsWi9y+DxBIKuFw8mPTtE7FFnFyo6+vD+vXr8f3v//9aSupCcCUJanjDV7VHAt4vV5UVlaiuLgYOTnR6mY+75SmaUEhkhOiSAtQspIaDkcWWYsRsBg9yDADf/o4QlABwOftH5PvQ0J0gaMRDM4c/nsQABAKsUIFf3l5eRIUShJZFOVdBYyKJDUcIkorAsR5eKFQCF1dXWhsbITP50NGRgacTifS0tI0CSvHcXC5WgDI7bTESqpKuH+yK6kSkhq5bliWRU1NDfR6PZYsWUI8Rtpm/sphZ47jYDQaUVBQgLS0AQDyWPF4pAnF9+Asr2Ynk1e5dReHK0/5m8ZnEoU4jzd2fVPLSQWAb397ABxnhU7nEz3EhwH8B1G3At4nVepTrKOjJJXjOFRWVgpuATzhIYf7o4PIw/2S8eMiqaRXldd5uSrJgy/+W7JkCex2O8LhbtKno3OjWKTbomuNWEk99fyNKD7pq4pzUMsbBQCbrLpfPLbBrN5YRa2dK03FphjxY0+0n+lIMTAwgPXr1+Oee+7B2rVrtT8wjaRjmqQmCWOVB9vd3Y3a2losWbJEoh7y+acAUFZWhn
```

Comment:
Finally, as part of this uniform structure, we want to always add a `What we learned` section at the end of the tutorial summarizing it in a couple of lines.

Reply:
Yes, again, let me ask @Zoufalc